torch/_tensor_str.py

```diff
@@ -46,12 +46,20 @@ def set_printoptions(
 
     Example::
 
+        >>> # Limit the precision of elements
         >>> torch.set_printoptions(precision=2)
         >>> torch.tensor([1.12345])
         tensor([1.12])
+        >>> # Limit the number of elements shown
         >>> torch.set_printoptions(threshold=5)
         >>> torch.arange(10)
         tensor([0, 1, 2, ..., 7, 8, 9])
+        >>> # Restore defaults
+        >>> torch.set_printoptions(profile='default')
```

Comment:
Something that is a bit bothersome is that if the doctest fails, the default restoration here will not occur, and then all subsequent tests will deliver nonsense. A "always run this afterwards no matter what" designation would be a big help.

Reply:
It is possible to use a try/finally in the doctest itself, but that get's a little ugly. This may be a good usecase for adding an `xdoctest: +finally` directive. But unfortunately there isn't a great way to accomplish doctest cleanup at the moment.